pkg/k8shandler/certificates.go

```diff
@@ -2,138 +2,81 @@ package k8shandler
+		return fmt.Errorf("Unable to get secret %s: %v", constants.MasterCASecretName, err)
 	}
-
-	value, ok := secret.Data[key]
-
-	// check to see if the map value exists
-	if !ok {
-		if deprecatedKeys.Has(key) {
-			log.Info("No secret data found. Please be aware but likely not an issue for deprecated keys", "key", key, "deprecatedKeys", deprecatedKeys.List())
-		} else {
-			log.Info("No secret data found", key)
-
+	workDir := utils.GetWorkingDir()
+	for name, value := range secret.Data {
+		if err != utils.WriteToWorkingDirFile(path.Join(workDir, name), value) {
+			return err
 		}
-		return nil
 	}
 
-	return utils.WriteToWorkingDirFile(toFile, value)
+	return nil
 }
 
 func (clusterRequest *ClusterLoggingRequest) writeSecret() (err error) {
 
+	secrets, err := loadFilesFromWorkingDir()
+	if err != nil {
+		return err
+	}
 	secret := NewSecret(
 		constants.MasterCASecretName,
 		clusterRequest.Cluster.Namespace,
-		map[string][]byte{
-			"masterca":  utils.GetWorkingDirFileContents("ca.crt"),
-			"masterkey": utils.GetWorkingDirFileContents("ca.key"),
-		})
-
+		secrets,
+	)
 	utils.AddOwnerRefToObject(secret, utils.AsOwner(clusterRequest.Cluster))
 
 	err = clusterRequest.CreateOrUpdateSecret(secret)
 	if err != nil {
 		return
 	}
-
 	return nil
 }
 
-func (clusterRequest *ClusterLoggingRequest) readSecrets() (err error) {
-
-	for secretName, certMap := range secretCertificates {
-		if err = clusterRequest.extractCertificates(secretName, certMap); err != nil {
-			return
-		}
+func loadFilesFromWorkingDir() (map[string][]byte, error) {
+	workDir := utils.GetWorkingDir()
+	files, err := ioutil.ReadDir(workDir)
+	if err != nil {
+		return nil, err
 	}
-
-	return nil
-}
-
-func (clusterRequest *ClusterLoggingRequest) extractCertificates(secretName string, certs map[string]string) (err error) {
-
-	for secretKey, certPath := range certs {
-		if err = clusterRequest.extractSecretToFile(secretName, secretKey, certPath); err != nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
-			return
-		}
+	results := map[string][]byte{}
+	for _, f := range files {
+		results[f.Name()] = utils.GetFileContents(path.Join(workDir, f.Name()))
 	}
-
-	return nil
+	return results, nil
 }
 
 //CreateOrUpdateCertificates for a cluster logging instance
 func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCertificates() (err error) {
-
-	// Pull master signing cert out from secret in logging.Spec.SecretName
-	if err = clusterRequest.readSecrets(); err != nil {
+	mutex.Lock()
+	defer mutex.Unlock()
+	if err = clusterRequest.extractMasterCerts(); err != nil {
+		fmt.Printf("Error %v", err)
 		return
 	}
 
 	scriptsDir := utils.GetScriptsDir()
-	if err = GenerateCertificates(clusterRequest.Cluster.Namespace, scriptsDir, "elasticsearch", utils.DefaultWorkingDir); err != nil {
+	if err = GenerateCertificates(clusterRequest.Cluster.Namespace, scriptsDir, "elasticsearch", utils.GetWorkingDir()); err != nil {
```

Comment:
If the script could indicate by its return value that it didn't change anything, we could skip writing the secret and cut down on API server traffic quite a bit.

Reply:
updated as we discussed